File: ColegioTrenerAndroid/app/src/main/java/com/jjmf/colegiotrenerandroid/data/dto/InscripcionDto.kt
package com.jjmf.colegiotrenerandroid.data.dto

import com.google.gson.annotations.SerializedName
import com.jjmf.colegiotrenerandroid.domain.model.Inscripcion

data class InscripcionDto(
    @SerializedName("tipoinscripcion") val tipoinscripcion: String?,
    @SerializedName("codtipoinscripcion") val codtipoinscripcion: String?,
    @SerializedName("inscripcion") val inscripcion: String?,
    @SerializedName("codinscripcion") val codinscripcion: String?,
    @SerializedName("precio") val precio: Double?,
    @SerializedName("estadoinscripcion") val estadoinscripcion: String?
) {
    fun toDomain(): Inscripcion {
        return Inscripcion(
            tipoinscripcion = tipoinscripcion?.trim(),
            codtipoinscripcion = codtipoinscripcion?.trim(),
            inscripcion = inscripcion?.trim(),
            codinscripcion = codinscripcion?.trim(),
            precio = precio ?: 0.0,
            estadoinscripcion = estadoinscripcion?.trim()
        )
    }
}

File: ColegioTrenerSwift/ColegioTrenerSwift/Network/Dto/InscripcionDto.swift
//
//  InscripcionDto.swift
//  ColegioTrenerSwift
//
//  Created by Joel Maldonado on 25/04/24.
//

import SwiftUI

struct InscripcionDto: Codable {
    let tipoinscripcion: String?
    let codtipoinscripcion: String?
    let inscripcion: String?
    let codinscripcion: String?
    let precio: Double?
    let estadoinscripcion: String?
    let inscripcionbloqueo: Double?

    func toDomain() -> Inscripcion {
        return Inscripcion(
            tipoinscripcion: tipoinscripcion?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            codtipoinscripcion: codtipoinscripcion?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            inscripcion: inscripcion?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            codinscripcion: codinscripcion?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            precio: precio ?? 0.0,
            estadoinscripcion: estadoinscripcion?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            inscripcionbloqueo: inscripcionbloqueo == 0.0
        )
    }
}
